Cq 4348367 changes for forms cc

barshat7 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

### Proxy Components
<img width="452" alt="image" src="https://user-images.githubusercontent.com/26343371/195773466-86dd0fdc-a32c-4642-90a5-309f51fa50b0.png">


## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Commit:
CQ-4348367 Changes for forms core components

Commit:
Merge branch 'develop' into CQ-4348367_changes_for_forms_cc

Commit:
CQ-4348367 adding core dependencies in all

Commit:
CQ-4348367 proxy components for wizard and title

nit23uec commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/template-types/af-page-v2_1/.content.xml:
```diff
@@ -0,0 +1,8 @@
+    <jcr:content
+        jcr:description="Generic template for Adaptive Form with core components V2.1"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Adaptive Form V2.1"/>
```
title needs to be updated.

barshat7 replied:
As per last discussion we are reusing the existing template & template types for headless .
cc :- @rismehta

nit23uec commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/template-types/af-page-v2_1/initial/.content.xml:
```diff
@@ -0,0 +1,16 @@
+        sling:resourceType="${appId}/components/page"
+        guideComponentType="fd/af/templates">
+        <guideContainer
+            fd:version="2.0"
```
fd:version="2.1" for everything related to core components.

barshat7 replied:
Done

nit23uec commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/templates/blank-af-v2_1/.content.xml:
```diff
@@ -0,0 +1,11 @@
+        cq:lastModifiedBy="admin"
+        cq:templateType="/conf/headless01/settings/wcm/template-types/af-core-component-page-v2"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Blank With Core Components V2.1"
```
title needs update

nit23uec commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/templates/blank-af-v2_1/initial/.content.xml:
```diff
@@ -0,0 +1,16 @@
+        sling:resourceType="${appId}/components/page"
+        guideComponentType="fd/af/templates">
+        <guideContainer
+            fd:version="2.0"
```
fd:version="2.1"

nit23uec commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/templates/blank-af-v2_1/policies/.content.xml:
```diff
@@ -0,0 +1,22 @@
+            jcr:primaryType="nt:unstructured"
+            sling:resourceType="wcm/core/components/policies/mapping"/>
+        <guideContainer
+            cq:policy="${appId}/components/adaptiveForm/formcontainer/default"
```
I'm not able to see this policy in this PR. can you pls point me to the correct place?

barshat7 replied:
https://github.com/adobe/aem-project-archetype/blob/f57c7f9448996eeeab9e68a719cd04d393f70bb0/src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/policies/.content.xml#L540

Commit:
CQ-4348367 using the same proxy formcontainer as headless for v2 container

Commit:
CQ-4348367 updating versions of wcm and forms components

nit23uec commented on src/main/archetype/pom.xml:
```diff
@@ -74,15 +74,15 @@
         <aio.runtime.auth>${env.AIO_RUNTIME_AUTH}</aio.runtime.auth>
 #end
-        <core.wcm.components.version>2.19.0</core.wcm.components.version>
+        <core.wcm.components.version>2.21.0</core.wcm.components.version>
```
if there's no requirement of update of wcm core, we can skip it.

rismehta replied:
This should not be changed.

nit23uec commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/adaptiveForm/accordion/_cq_template.xml:
```diff
@@ -0,0 +1,19 @@
+    fieldType="panel"
+    jcr:title="Accordion">
+    <item_1 jcr:primaryType="nt:unstructured"
+            jcr:title="item_1"
```
lets have` Item 1 ` as title instead of item_1.

nit23uec commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/adaptiveForm/accordion/_cq_template.xml:
```diff
@@ -0,0 +1,19 @@
+            layout="responsiveGrid"/>
+    <item_2 jcr:primaryType="nt:unstructured"
+            jcr:title="item_2"
+            name="item_2"
```
ditto for Item 2

rismehta commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/adaptiveForm/formcontainer/.content.xml:
```diff
@@ -3,5 +3,5 @@
     jcr:title="Adaptive Form Container component"
-    sling:resourceSuperType="core/fd/components/form/container/v1/container"
+    sling:resourceSuperType="core/fd/components/form/container/v2/container"
     componentGroup="${appTitle} - Adaptive Forms"/>
```
This should have a hidden component group

rismehta commented:
We also need to update template type and template

rismehta commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/adaptiveForm/accordion/.content.xml:
```diff
@@ -0,0 +1,7 @@
+    jcr:primaryType="cq:Component"
+    jcr:title="Adaptive Form Accordion (v1) component"
+    sling:resourceSuperType="core/fd/components/form/accordion/v1/accordion"
+    componentGroup="${appTitle} - Adaptive Forms"/>
```
Group name should be changed to `${appTitle} - Adaptive Form`

rismehta commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/adaptiveForm/fileinput/_cq_template.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
We also need to create proxies for forms and communication portal component in archetype. Name should be 

`${appTitle} - Forms and Communications Portal`

sakshi-arora1 reviewed (commented):
update the content.xml accordingly https://github.com/adobe/aem-project-archetype/blob/develop/src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/template-types/.content.xml

Commit:
CQ-4348367 review comments: changing cmp group, added formscommunicationportal components

Commit:
CQ-4348367 upgrade wcm examples if forms included, review comments closed

Commit:
CQ-4348367 Update thumbnails for af2

nit23uec commented on src/main/archetype/ui.content/src/main/content/jcr_root/conf/__appId__/settings/wcm/templates/.content.xml:
```diff
@@ -7,6 +7,9 @@
     <spa-page-template />
+#end
+#if ( $includeFormsheadless )
+    <blank-af-v2/>
```
blank-af-v2 is required for forms core components. it should be included irrespective of headless flag.

nit23uec commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/page/customfooterlibs.html:
```diff
@@ -20,6 +20,7 @@
 #end
 #if ( $includeFormsheadless == "y" )
     <sly data-sly-call="${symbol_dollar}{clientlib.js @ categories='${appId}.forms.react', async=true}"/>
+    <sly data-sly-test="${symbol_dollar}{!wcmmode.edit}" data-sly-call="${symbol_dollar}{clientlib.js @ categories='core.forms.components.runtime.all', async=true}"/>
```
clientlibs should be included outside of headless flag. if includeForms or includeFormsEnrollment or communication is true.

nit23uec commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/page/customheaderlibs.html:
```diff
@@ -20,6 +20,7 @@
 #end
 #if ( $includeFormsheadless == "y" )
     <sly data-sly-call="${symbol_dollar}{clientlib.css @ categories='${appId}.forms.react'}" />
+    <sly data-sly-test="${symbol_dollar}{!wcmmode.edit}" data-sly-call="${symbol_dollar}{clientlib.css @ categories='core.forms.components.runtime.all'}"/>
```
clientlibs should be included outside of headless flag. if includeForms or includeFormsEnrollment or communication is true.

Commit:
CQ-4348367 include core components if includeForms, use headless flag to only include headless spa

Commit:
CQ-4348367 check for cloud version

Commit:
CQ-4348367 check for cloud version..

Commit:
CQ-4348367 check for headless for react clientlibs

Commit:
CQ-4348367 template changes on par with core components it

Commit:
CQ-4348367 separate  adaptive form and portal components

Commit:
CQ-4348367 adaptive form submit button

Commit:
Merge branch 'develop' into CQ-4348367_changes_for_forms_cc

Commit:
CQ-4348367 associate af v2 canvas theme with template

Commit:
Merge branch 'develop' into CQ-4348367_changes_for_forms_cc

Commit:
CQ-4348367 Debug mvn build failure. Need to Revert

Commit:
CQ-4348367 Debug mvn build failure. Need to Revert

Commit:
Revert " CQ-4348367 Debug mvn build failure. Need to Revert"

This reverts commit 859be02115382ee920831fa9bebfe7d072b4d9ba.

Commit:
Revert "CQ-4348367 Debug mvn build failure. Need to Revert"

This reverts commit d4c6226dd60383bf7dd29cf1ecce91f68ff6056c.

Commit:
CQ-4348367 include core components in cloud only

vladbailescu committed:
Switch to ubuntu-20.04

Switch to ubuntu-20.04 as the update to 22.04 (https://github.blog/changelog/2022-11-09-github-actions-ubuntu-latest-workflows-will-use-ubuntu-22-04/) seems to be causing tests failures:
```
2023-01-10T12:53:01.6940591Z [INFO] [INFO] Compiling @angular/router/testing : es2015 as esm2015
2023-01-10T12:53:05.2477787Z [INFO] [INFO] 10 01 2023 12:53:05.245:INFO [karma-server]: Karma v5.2.3 server started at http://localhost:9876/
2023-01-10T12:53:05.2478369Z [INFO] [INFO] 10 01 2023 12:53:05.247:INFO [launcher]: Launching browsers ChromeHeadless with concurrency unlimited
2023-01-10T12:53:05.2497125Z [INFO] [INFO] (node:2292) [log4js-node-DEP0004] DeprecationWarning: Pattern %d{DATE} is deprecated due to the confusion it causes when used. Please use %d{DATETIME} instead.
2023-01-10T12:53:05.2498554Z [INFO] [INFO] (node:2292) [log4js-node-DEP0004] DeprecationWarning: Pattern %d{DATE} is deprecated due to the confusion it causes when used. Please use %d{DATETIME} instead.
2023-01-10T12:53:05.2505232Z [INFO] [INFO] 10 01 2023 12:53:05.250:INFO [launcher]: Starting browser ChromeHeadless
2023-01-10T12:53:05.2641898Z [INFO] [INFO] (node:2292) [log4js-node-DEP0004] DeprecationWarning: Pattern %d{DATE} is deprecated due to the confusion it causes when used. Please use %d{DATETIME} instead.
2023-01-10T12:53:07.0112167Z [INFO] [INFO] 10 01 2023 12:53:07.010:ERROR [launcher]: Cannot start ChromeHeadless
2023-01-10T12:53:07.0112824Z [INFO] [INFO] 	
2023-01-10T12:53:07.0113978Z [INFO] [INFO] DevTools listening on ws://127.0.0.1:9222/devtools/browser/b96a691d-ec85-4600-a87b-dc10ca506918
2023-01-10T12:53:07.0114595Z [INFO] [INFO] [0110/125305.697634:WARNING:gpu_process_host.cc(1265)] The GPU process has crashed 1 time(s)
2023-01-10T12:53:07.0162860Z [INFO] [INFO] [0110/125305.837689:WARNING:gpu_process_host.cc(1265)] The GPU process has crashed 2 time(s)
2023-01-10T12:53:07.0163597Z [INFO] [INFO] [0110/125305.968590:WARNING:gpu_process_host.cc(1265)] The GPU process has crashed 3 time(s)
2023-01-10T12:53:07.0166297Z [INFO] [INFO] [0110/125306.089230:WARNING:gpu_process_host.cc(1265)] The GPU process has crashed 4 time(s)
2023-01-10T12:53:07.0166760Z [INFO] [INFO] [0110/125306.209451:WARNING:gpu_process_host.cc(1265)] The GPU process has crashed 5 time(s)
2023-01-10T12:53:07.0167158Z [INFO] [INFO] [0110/125306.329345:WARNING:gpu_process_host.cc(1265)] The GPU process has crashed 6 time(s)
2023-01-10T12:53:07.0167866Z [INFO] [INFO] [0110/125306.329371:FATAL:gpu_data_manager_impl_private.cc(407)] GPU process isn't usable. Goodbye.
2023-01-10T12:53:07.0168464Z [INFO] [INFO] #0 0x5565114edb39 base::debug::CollectStackTrace()
```

Commit:
CQ-4348367 update adaptive form json path

Commit:
Merge branch 'develop' into CQ-4348367_changes_for_forms_cc

nit23uec commented on src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/adaptiveForm/checkboxgroup/.content.xml:
```diff
@@ -0,0 +1,7 @@
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:description="Adaptive Form Check Box Group (v1) component"
+    jcr:primaryType="cq:Component"
+    jcr:title="Adaptive Form Check Box Group (v1) component"
```
@barshat7 , do we show `v1` string in proxy components? I dont think so.

Commit:
CQ-4348367 remove unnecessary versioning & redundant names for proxy components

rismehta committed:
Merge branch 'develop' into CQ-4348367_changes_for_forms_cc
